### ISSUE TITLE
Github Auth Button disabled

### ISSUE THREAD

**Issue by ShalokShalom** [slapperwan/gh4a#818]
To me, the authorize button is disabled, right from the initial launch.

This here seems similar and is solved:
https://github.com/gitpoint/git-point/issues/674

**Comment by maniac103** [comment]
The difference to said issue is that we don't use a WebView internal to the app, but use an external web browser app to do the login.

What kind of browser do you use? Do you have Google Chrome installed (and thus the login is opening in a custom tab), or do you use some other browser?

**Comment by ShalokShalom** [comment]
Blackberrys build in browser in my Passport SE.

**Comment by ShalokShalom** [comment]
Is there any chance to let me choose the browser, instead using the default one?

**Comment by maniac103** [comment]
If Chrome is installed, it'll use custom tabs unconditionally, so the answer is 'no' in that case (but the page should work fine in Chrome). In all other cases, if you have multiple browsers installed, you should get prompted to choose the browser to use when starting the login process. If not, maybe you have chosen a default browser before and need to undo that. In stock Android, this setting can be found via Settings -> Apps -> gear button -> Browser

**Comment by ShalokShalom** [comment]
I can run Android apps, while I am on a completely different OS. :)

And all the other applications prompt me, except that ones for Github auth.

**Comment by smichel17** [comment]
Which OS? That's probably relevant info to figure out how to fix this.

**Comment by maniac103** [comment]
Maybe that other OS works differently in intent resolution, I don't know. The relevant code is here, including a comment on what it does: https://github.com/slapperwan/gh4a/blob/master/app/src/main/java/com/gh4a/utils/IntentUtils.java#L52

**Comment by ShalokShalom** [comment]
Well, I think its the same since it happens elsewhere on pure Android as well.
I dont depend on this app, so if you are fine with it: I am as well. 

Its OS 10 from Blackberry, yeah and it used the Android Runtime 4.3
The developer in the linked thread with the same issue simply decided to opt out from WebView completely 

The question might be, how often that happens without any issue report.
Its a webkit based browser, while once again:

In the linked thread does it happen to complete Android users with their build in webview also.

**Comment by maniac103** [comment]
I'm not fine with it, I simply don't know what I should/could do to fix it :-( The linked report had a supposed fix (https://github.com/gitpoint/git-point/pull/681), which doesn't work for us because we can't control the WebView.

I am aware of the issue in question (greyed out auth button) on older AOSP WebView builds, so probably Blackberry's browser is based on that ... but I still can't change that browser. I also can't debug why the intent resolution for browser selection (seemingly) doesn't work as expected on BBOS for obvious reasons either. I'm pretty sure the vast majority of Android users either uses Chrome or an Android version recent enough to get their WebView updated by Google nowadays, so it's likely not an issue for most of our users.

I thought of adding back support for authenticating via user name/password for other reasons (circumventing [the Oauth organization access restrictions](https://help.github.com/articles/about-oauth-app-access-restrictions/) which is a problem many users are not aware of) and maybe via plain token as well (#783), so that'll probably help you as well.

**Comment by maniac103** [comment]
> I think Forkhub does un/pw auth well -- it just uses them to create the auth token automatically, then throws them away once it has the token.

That's what we used to do (until #464), but a *lot* of people didn't trust in us not storing the password. Just check older Play Store comments :-(

**Comment by smichel17** [comment]
We could still offer both options.

**Comment by maniac103** [comment]
Yes, that's what I said in my larger comment above ;-)

**Comment by ShalokShalom** [comment]
I trust in you and other Android applications fully support the optional application choice here on this platform. Github clients/WebView requests seem to be the exception. :)

**Comment by Matrixcoffee** [comment]
I'm also stuck with this. I even installed a so-called "modern" web browser (IceCatMobile), but the button does not work in any browser I tried. +1 on _any_ alternative auth flow.

**Comment by ShalokShalom** [comment]
It is confirmed, from my side, that this is internally broken in Githubs application.
I already reported it several days ago. No response until

**Comment by ShalokShalom** [comment]
![img_20180225_081519](https://user-images.githubusercontent.com/6344099/36655547-9b4439f6-1ac3-11e8-903d-d82b5ed66e88.png)

**Comment by maniac103** [comment]
Added back user/password auth with https://github.com/slapperwan/gh4a/commit/270e952c968be1b2daf5e526b3e15110b4a2f8a9

**Comment by ShalokShalom** [comment]
Will you put an APK online? :)

**Comment by maniac103** [comment]
I can, but you'll have to reinstall the app (no update possible) because I don't have access to the official signing keys

**Comment by ShalokShalom** [comment]
Thanks a lot ^-^

**Comment by ShalokShalom** [comment]
Can I test this?

**Comment by smichel17** [comment]
We're waiting on @slapperwan to make a release. #800

In the meantime you could always build it yourself to test it (the debug version will install alongside the release version). Android studio is the easiest way to set that up, although you can do it with purely command line tools, too.

**Comment by ShalokShalom** [comment]
Thanks a lot, I will wait for the build. My special situation makes it cumbersome to setup and build Android APK's

**Comment by Tunous** [comment]
This is resolved now?

**Comment by ShalokShalom** [comment]
No. You can use the casual password/username login instead. Simply use the current release and scroll eventually down, its a bit hidden. :slightly_smiling_face:

**Comment by Tunous** [comment]
I'm just wondering whether I should close this issue or not since there doesn't seem to be any other solution...

**Comment by ShalokShalom** [comment]
This issue is still valid

**Comment by Matrixcoffee** [comment]
Just updated from f-droid, and I can now finally take this app for a test drive.

Posted from Octodroid :D

**Comment by ShalokShalom** [comment]
Its also possible to login from Gitpoint now: https://idk.tn/gitpoint-test.apk

In the same/similar way, simply find the 'Can't login' button on the button.

Interestingly, Gitpoint marked this issue as 'high priority' while it was much quicker implemented here in Octodroid. 

Gitpoint also said that they will release soon a new release with an updated GUI and this fix in.

Much fun :D
